lib/scalapackfx.fpp

```diff
@@ -1975,7 +1981,86 @@ contains
     call infog2l(grow, gcol, desc, mygrid%nrow, mygrid%ncol, mygrid%myrow,&
         & mygrid%mycol, lrow, lcol, rsrc, csrc)
 
-  end subroutine scalafx_infog2l
+  end subroutine scalafx_infog2l_single
+
+  !> Maps global positions in a distributed matrix to local one.
+  !!
+  !! \param mygrid  BLACS descriptor.
+  !! \param desc  Descriptor of the distributed matrix.
+  !! \param grow  Global row indices.
+  !! \param gcol  Global column indices.
+  !! \param lrow  Local row indices on output.
+  !! \param lcol  Local column indices on output.
+  !! \param rsrc  Rows of the process owning the local matrix.
+  !! \param csrc  Columns of the process owning the local matrix.
+  !! \param calcAllIndices  Whether to calculate all lrow and lcol,
+  !!     even if the current process does not own them. (default: true)
+  !!
+  subroutine scalafx_infog2l_array(mygrid, desc, grow, gcol,&
+      & lrow, lcol, rsrc, csrc, calcAllIndices)
+    type(blacsgrid), intent(in) :: mygrid
+    integer, intent(in) :: desc(DLEN_)
+    integer, intent(in) :: grow(:), gcol(:)
+    integer, intent(out), dimension(size(grow)) :: lrow, rsrc
+    integer, intent(out), dimension(size(gcol)) :: lcol, csrc
+    logical, intent(in), optional :: calcAllIndices
+
+    real(dp) :: invMB, invNB
+    integer :: ir, ic, rcheck, ccheck
+    integer, dimension(size(grow)) :: rblk
+    integer, dimension(size(gcol)) :: cblk
+    logical :: calcAllIndices_
+
+    invMB = 1.0_dp / real(desc(MB_), kind=dp)
+    invNB = 1.0_dp / real(desc(NB_), kind=dp)
+    rblk = (grow - 1) * invMB
+    cblk = (gcol - 1) * invNB
+
+    rcheck = mod(mygrid%myrow + mygrid%nrow - desc(RSRC_), mygrid%nrow)
+    ccheck = mod(mygrid%mycol + mygrid%ncol - desc(CSRC_), mygrid%ncol)
+
+    do ir = 1, size(grow)
+      rsrc(ir) = mod(rblk(ir) + desc(RSRC_), mygrid%nrow)
+    end do
+
+    do ic = 1, size(gcol)
+      csrc(ic) = mod(cblk(ic) + desc(CSRC_), mygrid%ncol)
+    end do
+
+    calcAllIndices_ = .true.
+    if (present(calcAllIndices)) then
+      calcAllIndices_ = calcAllIndices
+    end if
+
+    do ir = 1, size(grow)
+      if (calcAllIndices_ .or. mygrid%myrow == rsrc(ir)) then
+        lrow(ir) = (rblk(ir) / mygrid%nrow + 1) * desc(MB_) + 1
+        if (rcheck >= mod(rblk(ir), mygrid%nrow)) then
+          if (mygrid%myrow == rsrc(ir)) then
+            lrow(ir) = lrow(ir) + mod(grow(ir) - 1, desc(MB_))
+          end if
+          lrow(ir) = lrow(ir) - desc(MB_)
+        end if
```

Comment:
```suggestion
      if (mygrid%myrow == rsrc(ir)) then
        lrow(ir) = (rblk(ir) / mygrid%nrow) * desc(MB_) + mod(grow(ir) - 1, desc(MB_)) + 1
      end if
```
I think, this would be equivalent, if we assume, that `calcAllIndices_` is always `.false.`.